cirq-core/cirq/transformers/merge_single_qubit_gates.py

```diff
@@ -152,3 +156,257 @@ def merge_func(m1: 'cirq.Moment', m2: 'cirq.Moment') -> Optional['cirq.Moment']:
         deep=context.deep if context else False,
         tags_to_ignore=tuple(tags_to_ignore),
     ).unfreeze(copy=False)
+
+
+# ----------------------------------------------------------------------
+# Impl merge_single_qubit_gates_to_phxz_symbolized: Start
+# ----------------------------------------------------------------------
+
+
+def _values_of_sweep(sweep: Sweep, key: str | sympy.Symbol):
+    p = sympy.Symbol(key) if isinstance(key, str) else key
+    return [resolver.value_of(p) for resolver in sweep]
+
+
+def _merge_single_qubit_gates_to_phxz_symbolized(
```

Comment:
I think this method is useful on its own ... so it should be public

maybe even turn it into a class an split its logic across 2+ methods

Reply:
Done.